### DISCUSSION TITLE
Impossible to trash folders

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#2148] [Q&A] [answered]
Recently I noticed I cannot trash folders anymore, files are ok, just folders not,neither  empty or full.
![schermata-12-12-22-09](https://user-images.githubusercontent.com/10681413/145729929-667febd4-0c52-48b6-a6a2-2c19cdae629b.png)

Searching for "trash portal failed gvfs" brings up nothing.
* Affects all users
* deleted` ~/.local/share/Trash`, no change
*  thunar has the same issue

Debian bullseye, git checkout. Not sure if this an issue of LXQt. Ideas?

**Comment by tsujan** [comment]
Have never encountered this message. It may have something to do with GVFS because you said Thunar shows it too.

If I encountered the same issue, I wouldn't like someone to mention such a nasty possibility: Are you sure the file system isn't corrupted?

**Comment by tsujan** [comment]
Nasty possibilities aside, once I couldn't trash anything. No message was shown but the files were deleted without being trashed. As if there was no Trash.

Fortunately, I found the cause: Inside the Google drive, mounted by `rclone`, I had trashed a file. It was trashed successfully but the problem appeared when I unmounted it. So, I mounted it again, deleted its hidden trash folder, stopped pcmanfm-qt, stopped all GVFS processes and, finally, restarted pcmanfm-qt. The problem was solved.

After that, I never trash anything inside Google drive.

**Reply by stefonarch** [reply]
It's completely weird. I  noticed that I also cannot overwrite folders when moving:  first copy some folder to some location, then move the same folder, check "overwrite" and it shows "cannot move the directory over the directory". Will do a  file system check...

**Reply by stefonarch** [reply]
Same home directory/filesystem with manjaro no issue, except the latter which looks identical. 
Cannot move a directory if it already exist in target, only copy. This looks like a bug.

**Reply by tsujan** [reply]
Yes, it's weird. What about read-write permissions? Are they correct?

**Reply by tsujan** [reply]
Wait a minute: "Can’t move directory over directory" is a GLib message. GLib doesn't allow it in `glocalfile.c`. So, there's no bug anywhere.

However, the strange thing for me is that I'd never tried to move a dir to a dir before. Or, perhaps, I'd seen that message when I was a Gnome user and made a habit based on it?

**Reply by stefonarch** [reply]
Permissions are fine, and the other debian user has the same issue. 

But the impossibility to  overwrite folders looks not ok to me. Users can click "overwrite recursively" and it's still impossible.

**Reply by tsujan** [reply]
> But the impossibility to overwrite folders

Overwriting is possible; GLib doesn't allow *moving*.

**Reply by tsujan** [reply]
Of course, we can work around this limitation by overwriting the target and then removing the source folder — as we do when moving files to a different file system — but I don't think it's a serious limitation.

**Reply by tsujan** [reply]
Let's summarize it.

On the same file system, GLib doesn't allow

 * Moving of a directory over a directory with the same name;
 * Moving/copying of a file over a directory with the same name; and
 * Copying of a directory over a file with the same name.

But, apparently, it allows moving of a directory over a file with the same name. If something is strange, that's it. (I'm not 100% sure that it's GLib that allows the latter.)

**Comment by stefonarch** [comment, marked as answer]
Debian got an upgrade 11.1 → 11.2 so I decided to give it a look and booted into it. 
As said, searching for this error gives nearly zero results, but this "portal" in the name was a hint. 
https://issueexplorer.com/issue/flatpak/flatpak/4317#8303909
So I uninstalled all xdg-desktop-portals* (3 of them -gtk, -lxqt-git and no extension) and got on trashing folders this:
```
GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown:
 The name org.freedesktop.portal.Desktop was not provided by any .service files
```
Next step: controlling what I experimented with `GTK_USE_PORTAL=1` and voilà, I found that line in `/etc/env` (in manjaro it lead to firefox not opening a filedialog at all). Removing it, reinstalled `xdg-desktop-portal `and `xdg-desktop-portal-lxqt-git`, rebooted and trashing folders worked again.
